### PR TITLE
Refactors screen object update_icon to call parent

### PR DESCRIPTION
**Pull request by ninjanomnom**
And makes update_icon an /atom proc

This absolutely breaks things but I'm unable to run a test until later tonight

### CODE REVIEW

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -129,81 +129,85 @@
...
+				add_overlay(blocked_overlay)
+
+	if(held_index == hud.mymob.active_hand_index)
+		add_overlay("hand_active")
 
 
 /obj/screen/inventory/hand/Click(location, control, params)
 	// At this point in client Click() code we have passed the 1/10 sec check and little else
 	// We don't even know if it's a middle click
 	if(world.time <= usr.next_move)
```
Comment (Qustinnus) [inline code/_onclick/hud/screen_objects.dm]: Why not check my mob earlier and avoid using usr

File: code/_onclick/hud/screen_objects.dm [hunk 2]
```diff
@@ -532,26 +532,31 @@
 				return BODY_ZONE_HEAD
 
 /obj/screen/zone_sel/proc/set_selected_zone(choice, mob/user)
-	if(isobserver(user))
+	if(user != hud.mymob)
```
Comment (Qustinnus) [inline code/_onclick/hud/screen_objects.dm]: Why do you inconsistently check sanity on mymob? You do ?. In other procs

File: code/_onclick/hud/screen_objects.dm [hunk 3]
```diff
@@ -129,81 +129,85 @@
...
+		add_overlay("hand_active")
 
 
 /obj/screen/inventory/hand/Click(location, control, params)
 	// At this point in client Click() code we have passed the 1/10 sec check and little else
 	// We don't even know if it's a middle click
 	if(world.time <= usr.next_move)
-		return 1
+		return TRUE
 	if(usr.incapacitated() || isobserver(usr))
```
Comment (Qustinnus) [inline code/_onclick/hud/screen_objects.dm]: Usr!=mymob

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ninjanomnom** [comment]
Tested it locally and everything in the inventory seems to be fine, I expect some bugs will be found though

File: code/modules/cargo/gondolapod.dm [hunk 4]
```diff
@@ -32,7 +32,7 @@
 	name = linked_pod.name
 	. = ..()
 
-/mob/living/simple_animal/pet/gondola/gondolapod/proc/update_icon()
+/mob/living/simple_animal/pet/gondola/gondolapod/update_icon_state()
```
Comment (actioninja) [inline code/modules/cargo/gondolapod.dm]: shouldn't this be `update_icon()`?

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 5]
```diff
@@ -833,7 +833,7 @@ Pass a positive integer as an argument to override a bot's default speed.
 				ejectpai(usr)
 	update_controls()
 
-/mob/living/simple_animal/bot/proc/update_icon()
+/mob/living/simple_animal/bot/update_icon_state()
```
Comment (actioninja) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: ditto

**Comment by Anonmare** [comment]
This literally broke everything

**Comment by AarontheIdiot** [comment]
you just lost subscriber

**Pull request merged**